src/Symfony/OtelSdkBundle/DependencyInjection/Configuration/TraceConfiguration.php

```diff
@@ -0,0 +1,265 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Symfony\OtelSdkBundle\DependencyInjection\Configuration;
+
+use OpenTelemetry\Symfony\OtelSdkBundle\DependencyInjection\Configuration\Closure\ConfigurationExceptionCallback;
+use OpenTelemetry\Symfony\OtelSdkBundle\DependencyInjection\Configuration\Closure\CustomServiceValidatorCallback;
+use OpenTelemetry\Symfony\OtelSdkBundle\DependencyInjection\Configuration\Closure\ExporterConfigNormalizerCallback;
+use OpenTelemetry\Symfony\OtelSdkBundle\DependencyInjection\Configuration\Closure\IsNodeTypeCallback;
+use OpenTelemetry\Symfony\OtelSdkBundle\DependencyInjection\Configuration\Closure\IsOneOfNodeTypesCallback;
+use OpenTelemetry\Symfony\OtelSdkBundle\DependencyInjection\Configuration\Closure\LogicalEndCallback;
+use OpenTelemetry\Symfony\OtelSdkBundle\DependencyInjection\Configuration\Closure\StringToArrayCallback;
+use OpenTelemetry\Symfony\OtelSdkBundle\DependencyInjection\ConfigurationInterface;
+use OpenTelemetry\Symfony\OtelSdkBundle\Util\TreeBuilderHelper;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Config\Definition\Builder\NodeDefinition;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+
+class TraceConfiguration implements ConfigurationInterface
+{
+    use ConfigurationBehavior;
+
+    // PRIVATE CONSTANTS
+    private const SCALAR_NODE_TYPE = 'scalar';
+    private const PROCESSORS_XML = 'processor';
+    private const EXPORTERS_XML = 'exporter';
+    private const OPTIONS_XML = 'option';
+    private const ENV_PREFIX = 'env_';
+    private const EXPORTER_HR = 'span exporter';
+    private const PROCESSOR_HR = 'span processor';
+
+    /**
+     * @param bool $debug The kernel.debug value
+     */
+    public function __construct(bool $debug = false, ?LoggerInterface $logger = null)
+    {
+        $this->debug = $debug;
+        $this->logger = $logger;
+    }
+
+    public function getConfigTreeBuilder(): TreeBuilder
+    {
+        return TreeBuilderHelper::createArrayTreeBuilderAndAppend(
+            self::getName(),
+            self::canBeDisabled(),
+            self::createSamplerSectionNode(),
+            self::createSpanSectionNode(),
+            self::createExporters()
+        );
+    }
+
+    public static function getName(): string
+    {
+        return ConfigurationInterface::TRACE_NODE;
+    }
+
+    public static function canBeDisabled(): bool
+    {
+        return true;
+    }
+
+    private static function createSamplerSectionNode(): NodeDefinition
+    {
+        return TreeBuilderHelper::createArrayRootNode(self::SAMPLER_NODE)
+            ->beforeNormalization()
+            ->ifString()
+            ->then(
```

Comment:
this looks like some conditional methods - could it be indented to show that and aide readability? (if cs-fixer will allow it)

Reply:
It is, and no sadly CS is always removing my indentations